### PR TITLE
br: observe all of tables and support create/truncate table

### PR DESCRIPTION
**Pull request by joccau**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/32695
- hidden flag flagStreamFullBackupStorage when stream restore
- show progress and checkSum for restore ddl-events
- support restore ddl for new created db and table
- support restore partition table DML during stream restore
- support create/truncate table in PITR
- observer all data in stream task using hidden --filter when stream start

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [x] Performance regression: Consumes more CPU
- [x] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: br/pkg/backup/client.go [hunk 1]
```diff
@@ -405,12 +405,8 @@ func BuildBackupRangeAndSchema(
...
 				return nil, nil, nil, errors.Trace(err)
 			}
-			for _, r := range tableRanges {
-				ranges = append(ranges, rtree.Range{
-					StartKey: r.StartKey,
-					EndKey:   r.EndKey,
-				})
-			}
+
+			ranges = append(ranges, tableRanges...)
```
Comment (3pointer) [inline br/pkg/backup/client.go:409]: this logic seems rollback?

File: br/pkg/task/stream.go [hunk 2]
```diff
@@ -437,13 +461,27 @@ func RunStreamStart(
 			"and restart tikv")
 	}
 
-	if err = streamMgr.setGCSafePoint(ctx); err != nil {
+	if err := streamMgr.setGCSafePoint(ctx); err != nil {
```
Comment (3pointer) [inline br/pkg/task/stream.go:464]: 🤔

File: br/pkg/task/stream.go [hunk 3]
```diff
@@ -743,46 +768,48 @@ func RunStreamRestore(
...
+	// fix indices
 	pi := g.StartProgress(ctx, "Restore Index", countIndices(fullBackupTables), cfg.LogProgress)
 	err = withProgress(pi, func(p glue.Progress) error {
-		for _, table := range fullBackupTables {
-			if err := client.FixIndicesOfTable(ctx, table.DB.Name.L, table.Info, p.Inc); err != nil {
-				return errors.Annotatef(err, "failed to fix index for table %s.%s", table.DB.Name, table.Info.Name)
-			}
-		}
-		return nil
+		return client.FixIndicesOfTables(ctx, fullBackupTables, p.Inc)
```
Comment (3pointer) [inline br/pkg/task/stream.go:812]: if we backup full range. can we ignore this step?

File: br/pkg/restore/util_test.go [hunk 4]
```diff
@@ -268,7 +268,7 @@ func TestPaginateScanRegion(t *testing.T) {
 
 	_, err = restore.PaginateScanRegion(ctx, NewTestClient(stores, regionMap, 0), []byte{2}, []byte{1}, 3)
 	require.Error(t, err)
-	require.Regexp(t, ".*startKey >= endKey.*", err.Error())
+	require.Regexp(t, ".*startKey > endKey.*", err.Error())
```
Comment (3pointer) [inline br/pkg/restore/util_test.go:271]: 🤔

**Review by 3pointer — COMMENTED** [review]
reset LGTM

File: br/pkg/task/stream.go (hunk 3 above)
Comment (YuJuncen) [reply br/pkg/task/stream.go:812]: I guess maybe we need record whether indices are backed up in the backup meta...

**Review by YuJuncen — APPROVED** [review]

File: br/pkg/task/stream.go (hunk 3 above)
Comment (3pointer) [reply br/pkg/task/stream.go:963]: agree, @joccau you can leave a TODO here, and fix it in another PR.

**Pull request merged**